src/cmap/connection_pool.ts

```diff
@@ -354,41 +354,17 @@ export class ConnectionPool extends TypedEventEmitter<ConnectionPoolEvents> {
    * will be held by the pool. This means that if a connection is checked out it MUST be checked back in or
    * explicitly destroyed by the new owner.
    */
-  async checkOut(options?: { timeout?: Timeout }): Promise<Connection> {
+  async checkOut(options?: { timeoutContext?: TimeoutContext }): Promise<Connection> {
     this.emitAndLog(
```

Comment:
What issues does it cause to make both options and timeoutContext required?

If it's not too much of a hassle, I'd tackle it now - it feels like it defeats the purpose of having a legacy timeout context if we still need `Timeout.expires(this.options.waitQueueTimeoutMS);` here.

Reply:
We can make the options required without issue, so I'll do that here.
For making `timeoutContext` required, we'd have to change `CommandOptions` to require `timeoutContext` or change the type of the `options` parameter for `Server.command` to use a different type than `Connection.command`. The latter would be a band-aid fix until we got to implementing the socket timeouts for CSOT.